projects/addon-table/components/table/th/th.component.ts

```diff
@@ -52,6 +53,9 @@ export class TuiThComponent<T extends Partial<Record<keyof T, any>>> {
         @Optional()
         @Inject(forwardRef(() => TuiTableDirective))
         readonly table: TuiTableDirective<T> | null,
+        @Optional()
+        @Inject(forwardRef(() => TuiSortByDirective))
+        private readonly sortBy: TuiSortByDirective<T> | null,
```

Comment:
SortBy directive is kind of an extension over the regular table API to allow sorting by key, baseline table should not depend on it.

Reply:
Ok, I reverted `Th` :+1:

now I tried to invert the flow a little:  we can call `check` and update parent table directive from `TuiSortByDirective` setter, instead of relying on ngDoCheck